cli/src/funTest/assets/gradle-all-dependencies-expected-result-with-curations.yml

```diff
@@ -462,11 +462,15 @@ advisor: null
 evaluator: null
 resolved_configuration:
   package_curations:
-  - id: "Maven:org.hamcrest::"
-    curations:
-      comment: "Use the actual homepage instead of the GitHub page."
-      homepage_url: "http://hamcrest.org/JavaHamcrest/"
-  - id: "Maven:org.hamcrest:hamcrest-core:"
-    curations:
-      comment: "Fix description."
-      description: "Curated description."
+    providers:
+    - "File"
+    data:
+      File:
+      - id: "Maven:org.hamcrest::"
+        curations:
+          comment: "Use the actual homepage instead of the GitHub page."
+          homepage_url: "http://hamcrest.org/JavaHamcrest/"
+      - id: "Maven:org.hamcrest:hamcrest-core:"
+        curations:
+          comment: "Fix description."
+          description: "Curated description."
```

Comment:
Without having given it too much thought yet, I rather would have expected a structure like the following:

```
resolved_configuration:
  package_curations:
  - provider_id: "DefaultFile"
    curations:
    - id: "Maven:org.hamcrest::"
      curations:
        comment: "Use the actual homepage instead of the GitHub page."
        homepage_url: "http://hamcrest.org/JavaHamcrest/"
    - id: "Maven:org.hamcrest:hamcrest-core:"
      curations:
        comment: "Fix description."
        description: "Curated description."
```

Reply:
Hm, @mnonnenmacher, @MarcelBochtler, you an me discussed this in a recent developer meeting and what we came up with is this: https://github.com/oss-review-toolkit/ort/issues/6188#issuecomment-1383833350. (That's also mentioned in the commit message)

So, this PR is targeting this proposal, but leaving out the provider meta-data for now.

So, can we stick to it as-is?

Reply:
I'd be curious about the rationale that you guys discussed for the redundancy of listing the provider ids both inside `providers` *and* below `data`. Maybe I'm missing something, but this seems overly complicated to me.

Reply:
Ah, it looks so redundant because the providers in this example do not require any configuration. I guess the idea was to make `providers` inside `resolved_configuration` more or less a copy of `packageCurationProviders` from `config.yml`, right?

Reply:
> I'd be curious about the rationale that you guys discussed for the redundancy of listing the provider ids both inside providers and below data. Maybe I'm missing something, but this seems overly complicated to me.

The proposal was created in a discussion with you @sschuberth, @mnonnenmacher , @MarcelBochtler and me. 
That model wasn't, as far as I remember discussed any further without you being present. 

> Ah, it looks so redundant because the providers in this example do not require any configuration. I guess the idea was to make providers inside resolved_configuration more or less a copy of packageCurationProviders from config.yml, right?

Yes.

Reply:
@sschuberth and me agreed on discussing this in the @oss-review-toolkit/core-devs round.

Reply:
Outcome of discussion with @sschuberth @mnonnenmacher @MarcelBochtler : Let's change to:
```
resolved_configuration:
  package_curations:
  - provider:
      id: "CustomFile"
      type: "File"
      config:
        path: '/some-path/curations-dir'
        mustExist: false
    curations:
    - id: "Maven:org.hamcrest::"
      curations:
        comment: "Use the actual homepage instead of the GitHub page."
        homepage_url: "http://hamcrest.org/JavaHamcrest/"
    - id: "Maven:org.hamcrest:hamcrest-core:"
      curations:
        comment: "Fix description."
        description: "Curated description."
```

1. provider highest prio first
2. list of curations is ordered like the providers provided

Reply:
Implementation changed according to above comment.